### PR TITLE
Move a bunch of JS logic into CSS selectors

### PR DESCRIPTION
**Pull request by ap**
This is a continuation of #8.

### CODE REVIEW

File: src/content.js [hunk 1]
```diff
@@ -3,17 +3,19 @@
...
  // console.info({Found: elem});
 
- // Bail early if it's already one Chrome won't autodetect
- if( (!/^http/i.test(elem.getAttribute('action'))) &&
-     (!/^http/i.test(elem.action))
- ) {
-  return;
+ // Check whether the form submits to a HTTP(S) URL
+ var action = elem.action;
+ // If an input has name="action" it will mask the action property
```
Comment (cxw42) [inline src/content.js]: Do you have a specific failure case for the `getAttribute` check?  [This SO answer](https://stackoverflow.com/a/7590937/2877364) suggests that `getAttribute` handles the masking. 
 If we can use `getAttribute`, we won't need to touch the DOM.  However, I cheerfully yield to real testcases :) .

**Comment by cxw42** [comment]
@ap Thanks for doing this - push works fine now.  I will keep testing!

File: src/content.js (hunk 1 above)
Comment (ap) [reply src/content.js]: The point is that the `form.action` property is always a fully resolved absolute URL (well, except when it’s the empty string, in which case we get an equally fully-resolved absolute URL from `location.href`). By contrast, `form.getAttribute('action')` will yield whatever the HTML source contains, which can be a relative URL – so we have to resolve that value against `location.href` before we can look at it.

But, hmm. Now that I write this, it occurs to me that resolving the URL isn’t actually needed. If it’s a relative URL, then resolving it against `location.href` will always produce a URL the same scheme as `location.href`. So if it’s a relative URL, we don’t care what the absolute URL is, we just look at `location.href`.

Lemme just implement that…

Comment (ap) [reply src/content.js]: Oh wow that’s 15× better than before. Thanks for sending me back to the drawing board with this one.

**Comment by ap** [comment]
Hmm, updating my branch for the PR nukes your additions to it? :/

**Comment by ap** [comment]
Also it looks like GitHub now GCs commits super aggressively (didn’t used to be the case until not long ago… FOSTA?!), so even though I can still find the SHA1 of your commit I can no longer retrieve it. Oh well. Gotta be more careful next time…

**Comment by cxw42** [comment]
> Hmm, updating my branch for the PR nukes your additions to it? :/

Did you `push -f`?  On `fetch`, I got

    From https://github.com/ap/chrome-dont-add-custom-search-engines
     + 716ac1f...3d7f147 selectors  -> ap/selectors  (forced update)

I believe the workflow here is `fetch`/`commit`/`push` or `fetch`/`rebase`/`push`, rather than `push -f`.  However, this is actually the first time I have had the opportunity to work directly with someone on a PR, so I am not 100% sure.

I had:

    $ git status
    On branch selectors
    Your branch and 'ap/selectors' have diverged,
    and have 4 and 3 different commits each, respectively.

I tried rebasing but wound up with duplicate commits, so I just killed my local branch and checked out a fresh copy of yours.

> GC

I suppose aggressive GC is (in some ways) preferable to [BB's situation](https://bitbucket.org/site/master/issues/11593/allow-users-to-mark-repositories-for-git) :) .

**Comment by ap** [comment]
> I believe the workflow here is `fetch`/`commit`/`push` or `fetch`/`rebase`/`push`, rather than `push -f`.

You pushing commits to this PR doesn’t update the branch in my own repo, and I would be surprised if it did. So the `fetch` there does nothing.

Also, I rewrote one of my own earlier commits, so a force push was necessary.

And it would have been fine if I had your commit on my branch, but of course I didn’t, because it wasn’t on the branch in my repo. I thought GitHub would do some sort of magic or flag a conflict or something but nope… not a surprise exactly, of course. But… the whole thing does not strike me as super well thought out on GitHub’s part.

Oh well. I am guessing that I needed to manually fetch `refs/pull/10/head` from this repo into my local branch before rebasing it before pushing it to my fork.

**Comment by cxw42** [comment]
Weird - https://github.com/ap/chrome-dont-add-custom-search-engines/tree/selectors shows my commit (a424517) through the Web interface.  Do you see the same when you check it, or locally after a `fetch --all`?

**Comment by ap** [comment]
So it actually does update the branch in my own fork. Well that explains why deleting my fork prevented everybody else from being able to update the PR. And yeah, fetching from my fork updates the branch, as you should expect under those circumstances. That is all… a bit bizarre, frankly, but it may well be the only design that makes any sense given how GitHub has designed PRs.

I suppose that also explains what must have happened: I just went too quickly without thinking enough about what the messages I saw were telling me.

I guess I invited confusion upon my head because I generally use the original forked repo as my `origin` URL, and I only add my own fork as a `pushUrl`. That way fetches come from the original repo and pushes go to my fork – which is nice, except `refs/remotes/origin/` becomes a mixture of the refs in my fork and upstream. And I’m always juggling that discrepancy mentally, which distracted me from what I should have been paying attention to.

My first push after the rebase failed (because even though it was a force push, it was actually a `push --force-with-lease` (which I’ve aliased to `git force`)). So I flipped the URL to my own fork and did a `fetch --all`. I must have fetched your commit then, and just didn’t pay attention, when I should have expected it. I was too sure that *of course* the branch in my own fork couldn’t possibly be updated – even though it *had* to be for the force push to fail. And `--force-with-lease` is only a pretty weak protection: it only checks whether your remote tracking branch agrees with the corresponding ref on the remote. And `fetch --all` *makes* them all agree. So right after `fetch --all` it’s no better than regular `push -f`.

So the chain of events must have been that I did `push --force-with-lease`, then fetched your commit using `fetch --all`, but then didn’t rebase your commit onto my local branch (didn’t even notice it, frankly) and just blithely force-pushed my rebased branch, nuking your commit off of it in the process.

**Comment by ap** [comment]
Anyway, enough with the meta talk. Let’s get down to webextension business.

**Comment by cxw42** [comment]
@ap Sounds good :) .  Thanks for telling me about `--force-with-lease` - I didn't know about that option.  

I haven't seen any failures today with the latest code.  I want to run a few more testcases (and update the wiki to list them!) and then I think we'll be ready to merge.  The new `action` check does look quite slick :) .

**Comment by ap** [comment]
Ya I was quite please to discover that 😊 Thanks for the code review – without making me explain what the code was doing I wouldn’t have realised this much better possibility.

**Comment by cxw42** [comment]
@ap Looks good - I ran a bulk stress test [here](http://cxw42.github.io/chrome-dont-add.htm) (generated by [this](http://cxw42.github.io/chrome-dont-add.htm.in)) and the only ones it missed were those with ftp actions - which it should have missed :) .  I am going to merge this, then clean up the debug and indentation subsequently.  Thanks again!

**Pull request merged**